plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/PartitionTable.java

```diff
@@ -13,6 +13,7 @@
  */
 package io.trino.plugin.iceberg;
 
+import com.google.common.annotations.VisibleForTesting;
```

Comment:
does the cmt msg

> Reimplement equals in StructLikeWrapperWithFieldIdToIndex


describe the change?

let's use `Fix reading Iceberg $partitions table after updating partition schema` for commit title 
and add something like `Reimplement equals in StructLikeWrapperWithFieldIdToIndex` as the commit message (possibly explaining why it matters, or what was wrong)